tests/integration/test_transpose.py

```diff
@@ -12,25 +12,37 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+import numpy as np
 import pytest
 
 import cunumeric as num
 
+rect = num.array([[1, 2, 3], [4, 5, 6]])
+square = num.array([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
 
-def test_function():
-    x = num.array([[1, 2, 3], [4, 5, 6]])
-    y = num.transpose(x)
-    assert num.array_equal(y, [[1, 4], [2, 5], [3, 6]])
-    z = num.transpose(y)
-    assert num.array_equal(x, z)
 
+@pytest.mark.parametrize("x", (rect, square), ids=("rect", "square"))
+class Test_free_function:
+    def test_forward(self, x):
+        y = num.transpose(x)
+        assert num.array_equal(y, np.transpose(x))
```

Comment:
This is not checking anything. `np.transpose(x)` will go through NumPy's `__array_function__` mechanism, and eventually dispatch `num.transpose`. You instead probably want to do something like this:

```
np_x =  np.array([[1, 2, 3], [4, 5, 6]])
num_x = num.array(np_x)
assert num.array_equal(np.transpose(np_x), num.transpose(num_x))
```

Reply:
Ah, I thought it would go through `__array__`, good catch :+1:

Reply:
`x` is passed in as a `cunumeric` array to begin with, will this work?
```
npx = np.array(x)
assert num.array_equal(y, np.transpose(npx))
```

Reply:
@manopapad I've pushed a change inhttps://github.com/nv-legate/cunumeric/pull/457/commits/239849fda77ca18afcc646dec2fe5082c82501a2 let me know if this should suffice

Reply:
Yes, that will work.